axelor-account/src/main/java/com/axelor/apps/account/service/debtrecovery/DebtRecoveryService.java

```diff
@@ -358,6 +383,77 @@ else if (move.getInvoice() == null) {
     return query.fetch();
   }
 
+  /**
+   * Recovers all move lines for a specific partner, in the scope of the activities of a company,
+   * and optionally, for a specific trading name.
+   *
+   * @param partner A partner to be concerned by the move lines
+   * @param company A company to be concerned by the move lines
+   * @param tradingName (Optional) A trading name to be concerned by the move lines
+   * @return all corresponding move lines as a List
+   */
+  public List<InvoiceTerm> getInvoiceTerms(
+      Partner partner, Company company, TradingName tradingName) {
+
+    int mailTransitTime = company.getAccountConfig().getMailTransitTime();
+
+    Query<InvoiceTerm> query;
+    if (tradingName == null) {
+      query =
+          invoiceTermRepo
+              .all()
+              .filter(
+                  "(self.paymentSession IS NULL OR self.paymentSession.statusSelect != ?1) "
+                      + " and self.amountRemaining > 0 "
+                      + " and self.isPaid IS FALSE "
+                      + " and self.debtRecoveryBlockingOk IS FALSE "
+                      + " and self.moveLine IS NOT NULL "
+                      + " and self.moveLine.move.company = ?2 "
+                      + " and self.moveLine.partner = ?3 ",
+                  PaymentSessionRepository.STATUS_ONGOING,
+                  company,
+                  partner);
+    } else {
+      query =
+          invoiceTermRepo
+              .all()
+              .filter(
+                  "(self.paymentSession IS NULL OR self.paymentSession.statusSelect != ?1) "
+                      + " and self.amountRemaining > 0 "
+                      + " and self.isPaid IS FALSE "
+                      + " and self.debtRecoveryBlockingOk IS FALSE "
+                      + " and self.moveLine IS NOT NULL "
+                      + " and self.moveLine.move.company = ?2 "
+                      + " and self.moveLine.partner = ?3 "
+                      + " and self.moveLine.move.tradingName = ?4",
+                  PaymentSessionRepository.STATUS_ONGOING,
+                  company,
+                  partner,
+                  tradingName);
+    }
```

Comment:
Please refactor this query to avoid code duplication (construct a query, then add the trading name condition if the trading name is not null), also please use named parameters (ex: self.moveLine.partner = : partner) for better readability and avoiding future mistakes